src/PersistentFile.js

```diff
@@ -42,12 +33,11 @@ class PersistentFile extends EventEmitter {
     const json = {
```

Comment:
use `vfile`?

```js

const json = VFile({
  path: this.filepath,
  mtime: this.lastModifiedDate,
  originalName: this.originalFilename,
  mimetype: this.mimetype,
  size: this.size,
  hash: this.hashAlgorithm && this.hashAlgorithm !== '' ? this.hash : '',
});
```

Reply:
actually, this should be in the constructor 

```js
this._file = VFile(...)
```

then in `toJSON` just `return this._file`

Reply:
I feel like this PR is already too big, maybe do it in another ?

Reply:
yep.